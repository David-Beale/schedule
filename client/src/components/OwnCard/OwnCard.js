import React, { useState } from 'react';
import Moment from 'react-moment';
import Countdown from '../Countdown';
import EventModal from '../EventModal/EventModal';
import { DeleteOutlined } from '@ant-design/icons';
import './OwnCard.css';

function Card(props) {
  const [flipped, setFlipped] = useState(false);
  const [modalVisible, setModalVisible] = useState(false);
  const handleClick = () => {
    setFlipped(!flipped);
  };
  const toggleModal = (e) => {
    e.stopPropagation();
    setModalVisible(!modalVisible);
  };

  const handleChildClick = (e) => {
    e.stopPropagation();
  };

  return (
    <div>
      <div className="flip-card">
        <div
          onClick={handleClick}
          className={flipped ? 'flip flip-card-inner' : 'flip-card-inner'}
        >
          <div className="card-container flip-card-front">
            <div className="image-container">
              <img src={props.info.image} className="card-image" />
            </div>
            <div className="card-artist">{props.info.artistName}</div>
            <div className="card-time">
              <Moment format="ddd Do MMM, HH:mm">{props.info.date}</Moment>
            </div>
            <div className="card-title">{props.info.title}</div>
            <div
              onClick={handleChildClick}
              style={{
                textAlign: 'center',
                padding: '20px 0',
                width: '160px',
                height: '60px',
                borderRadius: '3px',
                backgroundColor: 'white'
              }}
            >
              <button className="delete-btn"
                onClick={() => props.handleDelete(props.id)}
                style={{ background: 'transparent', border: 'none' }}>
                <DeleteOutlined style={{ fontSize: "2em" }} />
              </button>
            </div>
          </div>

          <div className="card-container flip-card-back">
            <div className="card-title back">{props.info.title}</div>
            <div className="description-container">
              {props.info.description}
            </div>
            <div className="countdown-container">
              {flipped && <Countdown info={props.info} />}
            </div>
            <a onClick={toggleModal} className="event-link">
              Watch here!
            </a>
          </div>
        </div>
      </div>
      <EventModal
        visible={modalVisible}
        toggleModal={toggleModal}
        info={props.info}
      />
    </div>
  );
}
export { Card as default };
