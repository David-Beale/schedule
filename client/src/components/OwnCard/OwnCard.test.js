import OwnCard from './OwnCard';
import ReactDOM from 'react-dom';
import { render } from '@testing-library/react';
import React from 'react';

describe('OwnCard', () => {
  const RealDate = Date.now;

  beforeAll(() => {
    global.Date.now = jest.fn(() => new Date('2019-04-07T10:20:30Z'));
  });

  afterAll(() => {
    global.Date.now = RealDate;
  });
  it('should render without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<OwnCard info={{ image: '' }} />, div);
  });
  it('should render properly', () => {
    const component = render(<OwnCard info={{ image: '' }} />);
    expect(component).toMatchSnapshot();
  });
});
