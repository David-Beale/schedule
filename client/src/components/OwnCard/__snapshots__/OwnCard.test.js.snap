// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OwnCard should render properly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="flip-card"
        >
          <div
            class="flip-card-inner"
          >
            <div
              class="card-container flip-card-front"
            >
              <div
                class="image-container"
              >
                <img
                  class="card-image"
                  src=""
                />
              </div>
              <div
                class="card-artist"
              />
              <div
                class="card-time"
              >
                <time
                  datetime="1554632430000"
                >
                  Sun 7th Apr, 11:20
                </time>
              </div>
              <div
                class="card-title"
              />
              <div
                style="text-align: center; padding: 20px 0px; width: 160px; height: 60px; border-radius: 3px; background-color: white;"
              >
                <button
                  class="delete-btn"
                  style="background: transparent;"
                >
                  <span
                    aria-label="delete"
                    class="anticon anticon-delete"
                    role="img"
                    style="font-size: 2em;"
                  >
                    <svg
                      aria-hidden="true"
                      class=""
                      data-icon="delete"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M360 184h-8c4.4 0 8-3.6 8-8v8h304v-8c0 4.4 3.6 8 8 8h-8v72h72v-80c0-35.3-28.7-64-64-64H352c-35.3 0-64 28.7-64 64v80h72v-72zm504 72H160c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h60.4l24.7 523c1.6 34.1 29.8 61 63.9 61h454c34.2 0 62.3-26.8 63.9-61l24.7-523H888c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zM731.3 840H292.7l-24.2-512h487l-24.2 512z"
                      />
                    </svg>
                  </span>
                </button>
              </div>
            </div>
            <div
              class="card-container flip-card-back"
            >
              <div
                class="card-title back"
              />
              <div
                class="description-container"
              />
              <div
                class="countdown-container"
              />
              <a
                class="event-link"
              >
                Watch here!
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="flip-card"
      >
        <div
          class="flip-card-inner"
        >
          <div
            class="card-container flip-card-front"
          >
            <div
              class="image-container"
            >
              <img
                class="card-image"
                src=""
              />
            </div>
            <div
              class="card-artist"
            />
            <div
              class="card-time"
            >
              <time
                datetime="1554632430000"
              >
                Sun 7th Apr, 11:20
              </time>
            </div>
            <div
              class="card-title"
            />
            <div
              style="text-align: center; padding: 20px 0px; width: 160px; height: 60px; border-radius: 3px; background-color: white;"
            >
              <button
                class="delete-btn"
                style="background: transparent;"
              >
                <span
                  aria-label="delete"
                  class="anticon anticon-delete"
                  role="img"
                  style="font-size: 2em;"
                >
                  <svg
                    aria-hidden="true"
                    class=""
                    data-icon="delete"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M360 184h-8c4.4 0 8-3.6 8-8v8h304v-8c0 4.4 3.6 8 8 8h-8v72h72v-80c0-35.3-28.7-64-64-64H352c-35.3 0-64 28.7-64 64v80h72v-72zm504 72H160c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h60.4l24.7 523c1.6 34.1 29.8 61 63.9 61h454c34.2 0 62.3-26.8 63.9-61l24.7-523H888c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zM731.3 840H292.7l-24.2-512h487l-24.2 512z"
                    />
                  </svg>
                </span>
              </button>
            </div>
          </div>
          <div
            class="card-container flip-card-back"
          >
            <div
              class="card-title back"
            />
            <div
              class="description-container"
            />
            <div
              class="countdown-container"
            />
            <a
              class="event-link"
            >
              Watch here!
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
